src/pybind/mgr/dashboard/frontend/src/app/shared/directives/auth-storage.directive.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { Component } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import _ from 'lodash';
+
+import { Permission } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { configureTestBed } from '~/testing/unit-test-helper';
+import { AuthStorageDirective } from './auth-storage.directive';
+
+@Component({
+  template: `
+    <div *cdAuthStorage="'osd'">This displays if string</div>
+
+    <div *cdAuthStorage="['configOpt', 'osd', 'rgw']">This displays if Array</div>
+
+    <div *cdAuthStorage="{ rgw: ['read', 'update'], osd: ['read'] }">This displays if object</div>
+  `
+})
+export class TestComponent {}
+
+describe('AuthStorageDirective', () => {
+  let fakePermissions: Permission;
+  let authStorageService: AuthStorageService;
+  let directive: AuthStorageDirective;
+  let fixture: ComponentFixture<TestComponent>;
+
+  configureTestBed({
+    declarations: [AuthStorageDirective, TestComponent]
+  });
+
+  beforeEach(() => {
+    directive = new AuthStorageDirective(null, null, new AuthStorageService());
+
+    fakePermissions = {
+      create: false,
+      update: false,
+      read: false,
+      delete: false
+    };
```

Comment:
In order to quickly test multiple combinations of permissions, you could try [parameterized testing](https://jdkandersson.com/2019/06/30/parametrize-angular-jasmine-tests/) (like [we did here](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.spec.ts)).